blocks/text.js

```diff
@@ -825,18 +830,24 @@ Blockly.Constants.Text.TEXT_CHARAT_MUTATOR_MIXIN = {
         this.appendDummyInput('ORDINAL')
             .appendField(Blockly.Msg.ORDINAL_NUMBER_SUFFIX);
       }
-    } else {
-      this.appendDummyInput('AT');
     }
     if (Blockly.Msg.TEXT_CHARAT_TAIL) {
       this.removeInput('TAIL', true);
       this.appendDummyInput('TAIL')
           .appendField(Blockly.Msg.TEXT_CHARAT_TAIL);
     }
-    var menu = new Blockly.FieldDropdown(this.WHERE_OPTIONS, function(value) {
+
+    this.isAt_ = isAt;
+  }
+};
+
+// Does the initial mutator update of text_charAt and adds the tooltip
+Blockly.Constants.Text.TEXT_CHARAT_EXTENSION = function() {
+    var dropdown = this.getField('WHERE');
+    dropdown.setValidator(function(value) {
       var newAt = (value == 'FROM_START') || (value == 'FROM_END');
       // The 'isAt' variable is available due to this function being a closure.
```

Comment:
This comment is no longer true, since it's stored on the block.

Reply:
Done.